canonicalwebteam/discourse/parsers/engage.py

```diff
@@ -155,3 +157,103 @@ def _parse_related(self, tags):
         """
         index_list = [item for item in self.metadata if item["tags"] in tags]
         return index_list
+
+    def _parse_engage_metadata(self, index_soup, section_name):
```

Comment:
This looks to me like it's just a duplicate of `BaseParser._parse_metadata`, with your error handling around it. How is it different from the other function? Why didn't you just add your error handling into the basic metadata parser?